hibernate-core/src/test/java/org/hibernate/test/interceptor/InterceptorNonNullTransactionTest.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.interceptor;
+
+import java.util.Arrays;
+import java.util.Map;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.transaction.Status;
+
+import org.hibernate.EmptyInterceptor;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.resource.transaction.spi.TransactionStatus;
+
+import org.hibernate.testing.jta.TestingJtaBootstrap;
+import org.hibernate.testing.jta.TestingJtaPlatformImpl;
+import org.hibernate.testing.junit4.CustomParameterized;
+import org.hibernate.test.exceptionhandling.BaseJpaOrNativeBootstrapFunctionalTestCase;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(CustomParameterized.class)
+public class InterceptorNonNullTransactionTest extends BaseJpaOrNativeBootstrapFunctionalTestCase {
+
+	public enum JpaComplianceTransactionSetting { DEFAULT, TRUE, FALSE }
+	public enum JtaAllowTransactionAccessSetting {
+		DEFAULT {
+			@Override
+			public boolean allowTransactionAccess(
+					SessionFactory sessionFactory,
+					JpaComplianceTransactionSetting jpaComplianceTransactionSetting) {
+				return !sessionFactory.getSessionFactoryOptions().isJpaBootstrap() ||
+						jpaComplianceTransactionSetting != JpaComplianceTransactionSetting.TRUE;
+			}
+		},
+		TRUE {
+			@Override
+			public boolean allowTransactionAccess(
+					SessionFactory sessionFactory,
+					JpaComplianceTransactionSetting jpaComplianceTransactionSetting) {
+				return true;
+			}
+		},
+		FALSE {
+			@Override
+			public boolean allowTransactionAccess(
+					SessionFactory sessionFactory,
+					JpaComplianceTransactionSetting jpaComplianceTransactionSetting) {
+				// setting is ignored if jpaComplianceTransactionSetting != JpaComplianceTransactionSetting.TRUE
+				return jpaComplianceTransactionSetting != JpaComplianceTransactionSetting.TRUE;
+			}
+		};
+
+		public abstract boolean allowTransactionAccess(
```

Comment:
As far as I can tell, this method and its implementations are dead code. Bad copy/paste, maybe?

Reply:
I force-pushed an update to fix that. Will wait for CI and merge.